### ISSUE TITLE
ceno-node and plugin-freenet integration issues

### ISSUE THREAD

**Issue by misaakidis** [equalitie/ceno#13]
Following in comments reproducible bugs on the integration of the ceno-node.js and the freenet plugin.

**Comment by misaakidis** [comment]
cenotransport abnormally terminates upon {"bundleFound": "false"} (Content type "application/javascript") response to lookup requests.

------
Got request on route /
No existing process to bundle www.google.com
Requesting localhost:3091/?url=www.google.com succeeded.

undefined:1
undefined
^
SyntaxError: Unexpected token u
    at Object.parse (native)
    at /home/.../ceno-node/lib/cache.js:101:25

**Comment by misaakidis** [comment]
No "Please wait" page is shown to the user during lookup.

**Comment by misaakidis** [comment]
plugin-ceno returns {"bundleFound": "false"} when lookup does not find the requested resource in the cache.
ceno-node check of condition(!response.bundleFound) [line 103] is always true even when bundle was not found, because bundleFound value is the string "false" and not the boolean value false.

**Comment by redwire** [comment]
Plugin-ceno should produce {"bundleFound": false} and not {"bundleFound": "false"}.

**Comment by misaakidis** [comment]
transport-node.js line 47 catches err ECONREFUSED; plugin-ceno gets no post requests on :3092

---
Error: connect ECONNREFUSED
    at errnoException (net.js:904:11)
    at Object.afterConnect [as oncomplete] (net.js:895:19)

**Comment by misaakidis** [comment]
transport-node.js sends a post request to path "/" instead of "/store"

**Comment by misaakidis** [comment]
plugin-ceno cache handler gets url post value from request attributes, ceno-node passes it within a JSON object

**Comment by misaakidis** [comment]
Processes stored from previous executions of ceno-node "interfere" with new ones for the same url.

Requests get back as response the corresponding pid of the previous execution and no new process is started to get the bundle. In client side, every time the page refreshes, the please wait page is shown.

Proposed solution: clear db/processes.json at ceno-node initialization.

**Comment by misaakidis** [comment]
plugin-ceno CacheInsertHandler does not catch exception if POST request contains no body

**Comment by redwire** [comment]
As a suggestion, could you consider opening new issues for each of these problems so that we can discuss them individually and also keep track of which ones you've already fixed?

**Comment by misaakidis** [comment]
Following @redwire's suggestion, I'm splitting this issue into ones for each bug, marked with the label freenet-integration, and update their status there. Follow them by visiting https://github.com/equalitie/ceno/labels/freenet-integration

**Comment by redwire** [comment]
@misaakidis:

> Processes stored from previous executions of ceno-node "interfere" with new ones for the same url.

Fixed in https://github.com/equalitie/ceno/commit/6fac7ff53cc3d5280568c519dc6266a0a554c05f
I seem to have miswrote the commit message.  As you can see, I do indeed remove all processes.

> transport-node.js sends a post request to path "/" instead of "/store"

I'm a little unsure of how this could be occurring. Is the following not the correct configuration?
I'm not sure if this is as it is because you already fixed it or not.  Please let me know!

```javascript
var cacheServer = 'http://localhost:3092';
var cenoClient = 'http://localhost:3090';

/* Have the cache server store a bundle for a given url.
 */
function cacheBundle(url, bundle, callback) {
  console.log('Sent post request to ' + cacheServer);
  request
  .post(cacheServer + '/store')
  .send({url: url, bundle: bundle})
  .end(callback);
}
```

**Comment by misaakidis** [comment]
Yes I fixed it in 85ca8fbc
More on issue #18

**Comment by redwire** [comment]
Perfect!

I'm going to close this issue now so that we can focus the discussion on the more specific issues.
